DOCS-70: Add pgp key verification instructions to install chapter (#1616)
freenas/freenas-docs: userguide/install.rst

   tmoore@Observer ~/Downloads> gpg --keyserver sks-keyservers.net --recv-keys 0xc8d62def767c1db0dff4e6ec358eaa9112cf7946
   gpg: /usr/home/tmoore/.gnupg/trustdb.gpg: trustdb created
   gpg: key 358EAA9112CF7946: public key "IX SecTeam <security-officer@ixsystems.com>" imported
   gpg: Total number processed: 1
   gpg:               imported: 1
   tmoore@Observer ~/Downloads>

* Use :command:`gpg --verify` to compare the :file:`.iso` and
  :file:`.iso.gpg` files:

.. code-block:: none

   tmoore@Observer ~/Downloads> gpg --verify FreeNAS-11.2-U6.iso.gpg FreeNAS-11.2-U6.iso
   gpg: Signature made Tue Nov  5 13:48:18 2019 EST
   gpg:                using RSA key C8D62DEF767C1DB0DFF4E6EC358EAA9112CF7946
   gpg: Good signature from "IX SecTeam <security-officer@ixsystems.com>" [unknown]
   gpg: WARNING: This key is not certified with a trusted signature!
   gpg:          There is no indication that the signature belongs to the owner.
   Primary key fingerprint: C8D6 2DEF 767C 1DB0 DFF4  E6EC 358E AA91 12CF 7946
   tmoore@Observer ~/Downloads>

* This response means the signature is correct but still untrusted. Go
  back to the browser page that has the :guilabel:`PGP Public key` open

>   and manually confirm that the iX Security Team has signed the key.

Reviewer: Technically yes, but the iX Security Team self-signed the key. I would go with something like this: "that the key was issued for the iX Security Team and is signed by other iXsystems accounts", then the issue date must be October 15th, 2019.

Author: Ok, I've integrated your suggestions.

-------------------------

The latest STABLE version of %brand% |release| is available for download
from `<https://www.freenas.org/download-freenas-release/>`__.

The download page has links to %brand% release notes, :file:`.iso`
integrity checksums, and PGP security keys.

Clicking :guilabel:`Download` opens a dialog to save an *.iso* file.

> This bootable installer should be

Reviewer 2: ```suggestion
This bootable installer must be
```

Author: Not technically true, but w/e.

> :ref:`written to physical media <Preparing the Media>` before installing

Reviewer 2: ```suggestion
:ref:`written to physical media <Preparing the Media>` before it can be used to install
```

%brand%.

.. index:: Verify download files

Checking Installer Integrity
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

%brand% uses the
`OpenPGP standard <https://en.wikipedia.org/wiki/Pretty_Good_Privacy#OpenPGP>`__

> to show that downloadable files have been provided by a trustworthy

Reviewer 2: ```suggestion
to verify that downloaded files have been provided by a trustworthy
```

> source. Verifying the PGP signature of a %brand% file requires OpenPGP

Reviewer 2: Non-specific "requires".  Rewrite to be specific:
```suggestion
source. OpenPGP software like gnupg, Kleopatra, or Gpg4win can verify
the PGP signature of a %brand% installer file.
```

compliant encryption software like
`gnupg <https://www.freebsd.org/cgi/man.cgi?query=gpg>`__,
`Kleopatra <https://www.openpgp.org/software/kleopatra/>`__,
or `Gpg4win <https://gpg4win.org/>`__.

> The :file:`sha256.txt` file is used to confirm the integrity of the

Reviewer 2: If this is an introduction to the usage of the sha256 file, it should mention that details are shown below.  Otherwise, it can be integrated with that section.

downloaded :file:`.iso`.

PGP Verification
^^^^^^^^^^^^^^^^

To verify the :file:`.iso` source, go to
`<https://www.freenas.org/download-freenas-release/>`__ and click
:guilabel:`PGP Signature` to download the software signature file. Open
the :guilabel:`PGP Public key` link and note the browser address and
:literal:`Search results` string.

> Using any preferred OpenPGP compliant encryption software, import the

Reviewer 2: ```suggestion
Using any of the OpenPGP encryption tools mentioned above, import the
```

Clicking :guilabel:`Download` opens a dialog to save an *.iso* file.
This bootable installer must be
:ref:`written to physical media <Preparing the Media>` before it can be
used to install %brand%.

.. index:: Verify download files

Checking Installer Integrity
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

%brand% uses the
`OpenPGP standard <https://en.wikipedia.org/wiki/Pretty_Good_Privacy#OpenPGP>`__
to confirm that downloaded files have been provided by a trustworthy
source. OpenPGP compliant software like
`gnupg <https://www.freebsd.org/cgi/man.cgi?query=gpg>`__,
`Kleopatra <https://www.openpgp.org/software/kleopatra/>`__,
or `Gpg4win <https://gpg4win.org/>`__. can check the PGP signature of a
%brand% installer file.

The :file:`sha256.txt` file is used to confirm the integrity of the
downloaded :file:`.iso`. See :ref:`SHA256 Verification` for more
details.

PGP Verification
^^^^^^^^^^^^^^^^

To verify the :file:`.iso` source, go to
`<https://www.freenas.org/download-freenas-release/>`__ and click
:guilabel:`PGP Signature` to download the software signature file. Open
the :guilabel:`PGP Public key` link and note the browser address and
:literal:`Search results` string.

> Use one of the OpenPGP encryption tools that were mentioned above to

Reviewer 2: ```suggestion
Use one of the OpenPGP encryption tools mentioned above to
```

   gpg:               imported: 1
   tmoore@Observer ~/Downloads>

* Use :command:`gpg --verify` to compare the :file:`.iso` and
  :file:`.iso.gpg` files:

.. code-block:: none

   tmoore@Observer ~/Downloads> gpg --verify FreeNAS-11.2-U6.iso.gpg FreeNAS-11.2-U6.iso
   gpg: Signature made Tue Nov  5 13:48:18 2019 EST
   gpg:                using RSA key C8D62DEF767C1DB0DFF4E6EC358EAA9112CF7946
   gpg: Good signature from "IX SecTeam <security-officer@ixsystems.com>" [unknown]
   gpg: WARNING: This key is not certified with a trusted signature!
   gpg:          There is no indication that the signature belongs to the owner.
   Primary key fingerprint: C8D6 2DEF 767C 1DB0 DFF4  E6EC 358E AA91 12CF 7946
   tmoore@Observer ~/Downloads>

* This response means the signature is correct but still untrusted. Go
  back to the browser page that has the :guilabel:`PGP Public key` open
  and manually confirm that the key was issued for the iX Security Team
  on October 15, 2019 and has been signed by iXsystems accounts.

.. _SHA256 Verification:

SHA256 Verification
^^^^^^^^^^^^^^^^^^^

The command to verify the checksum varies by operating system:

* on a BSD system use the command

>   :samp:`sha256 {name_of_file}`

Reviewer 2: This can be more helpful to the reader by telling them which file:
```suggestion
  :samp:`sha256 {isofile}`
```

Author: This is technically out of scope for this PR/ticket.

  `HashTab <http://implbits.com/products/hashtab/>`__.

The value produced by running the command must match the value shown

> in the :file:`sha256.txt` file. Checksum values that do not match

Reviewer 2: ```suggestion
in the :file:`sha256.txt` file. Different checksum values
```
